src/citrine/resources/project.py

```diff
@@ -136,7 +142,7 @@ def predictor_evaluation_executions(self) -> PredictorEvaluationExecutionCollect
     @property
     def design_workflows(self) -> DesignWorkflowCollection:
         """Return a collection representing all visible design workflows."""
-        return DesignWorkflowCollection(self.uid, self.session)
+        return DesignWorkflowCollection(project_id=self.uid, branch_id=None, session=self.session)
```

Comment:
I ended up not deprecating this because `/api/v2/design-workflows` accepts an optional `branch` query parameter for filtering. Omitting it gets them all the (unarchived) design workflows in the project, just like this property currently does. So it seems reasonable to allow a user to invoke that by calling `Project.design_workflows`.

However, that also means that until the v1 design workflow endpoints are deprecated, the user cannot actually invoke that form. They either call `Project.design_workflows`, which gets all the design workflows but doesn't include their branch IDs, or they call `Branch.design_workflows`, which has their branch IDs but only gets design workflows from that branch.

That being said, I don't think it causes any practical issues, as I have a hard time imagining a real-world flow where a user is hunting for a specific design workflow just to get the branch ID.

Reply:
Conversing with James and Andy, the solution we settled on is to just always hit v2. What this means is that to create a design workflow without a branch ID, we'll hit the v1 API, then retrieve the branch ID with a get.